### ISSUE TITLE
<pre>-element changes to innerHTML not shown until execution ends

### ISSUE THREAD

**Issue by darth83** [skulpt/skulpt#1197]
Using a simple gist for Skulpt: https://gist.github.com/bnmnetp/4650616#file-simpleskulpt-html

When you run a simple python script like:
```
print('Hello')
x = input()
print('Nice to meet you', x)
```

you can see that innerHTML of the element is updated (in dev tools) but the changes are not shown on the page until the script has ended. I'm using Chrome (same issue on iPad Safari). It works in Firefox though. Does anybody know how this problem is solved at trinket.io and https://py3.codeskulptor.org/ where the issue doesn't exist?

To sum up, what I want is that the output in the \<pre>-element is updated continuously and not only when the script has ended. What I have understood is that browsers are not required to show changes to an element while the javascript is running.

**Comment by s-cork** [comment]
you could try some of the flags in the `Sk.compile` options

I think `killableWhile: true, killableFor: true` might be the ones you need to allow the print statements to run before the script has ended.

**Comment by darth83** [comment]
I have been experimenting with all options that I think could be related. Unfortunately it doesn't help setting killableWhile or/and killableFor to true. Any other suggestion?

**Comment by s-cork** [comment]
here's a jsfiddle that's similar (i think) to the way trinket does it.

The key seems to be the extra argument to handle what happens during suspensions. 
https://jsfiddle.net/sc549/bjLuq5s7/

```javascript
  Sk.misceval.asyncToPromise(() =>
    Sk.importMainWithBody("<stdin>", false, prog, true), {
      // handle a suspension of the executing code
      // "*" says handle all types of suspensions
      "*": () => {
        if (StopExecution) throw "Execution interrupted"
      }
    },
  ).catch(err => {
    outf("\n" + err.toString());
  }).finally(() => {
    run.disabled = false;
  })
}
```

**Comment by darth83** [comment]
Thanks for trying to help out. How should I use this for my issue above? 

The jsfiddle only shows the ability to stop execution, not pause (which I think is necessary to allow the browser to update elements), and it doesn't print before an input().

**Comment by ebertmi** [comment]
I have just recreated your example in a jsFiddle (https://jsfiddle.net/hro6n318/1/) and I cannot reproduce your problem in Chrome (85.0.4183.121). Which version are you using? It makes me suspicious that it works in Firefox and not in Chrome.

**Comment by darth83** [comment]
> How about something like this
> 
> ```js
> Sk.configure({
>   inputTakesPrompt: true,
>   inputfun(args) {
>     return new Promise((resolve, reject) => {
>       setTimeout(function() {
>         resolve(window.prompt(args))
>       }, 250)
>     })
>   },
>   __future__: Sk.python3
> });
> ```

This works, thanks! Unfortunately it raises another problem, using this, the time the input box is open counts towards execLimit. Any way to stop that?

**Comment by darth83** [comment]
> I have just recreated your example in a jsFiddle (https://jsfiddle.net/hro6n318/1/) and I cannot reproduce your problem in Chrome (85.0.4183.121). Which version are you using? It makes me suspicious that it works in Firefox and not in Chrome.

I was actually using an older version of Chrome but now I have updated to the same as you and the problem is still there, in your fiddle as well. (I'm using Linux). However, the issue is there for Safari on iPad as well.

**Comment by s-cork** [comment]
>  counts towards execLimit. Any way to stop that?
You could remove `execLimit` from the options it was mostly there to server as an example of some of the available options.

`killableWhile` and `killableFor` with the option of how to handle suspensions included in `Sk.misceval.asyncToPromise` is enough to not lock up the browser and kill infinite loops. I've updated the fiddle.

**Comment by darth83** [comment]
> You could remove `execLimit` from the options it was mostly there to server as an example of some of the available options.

Yes, of course. But my preferred way would be that I use execLimit, but it is not super important, a stop button should work as well.

**Comment by s-cork** [comment]
You could do this
```javascript
Sk.configure({
  inputTakesPrompt: true,
  inputfun(args) {
    return new Promise((resolve, reject) => {
      setTimeout(function() {
        resolve(window.prompt(args))
      }, 250).then(() => {Sk.execStart = new Date()}) // reset the execution start time
    })
  },
  __future__: Sk.python3
});

**Comment by darth83** [comment]
I doesn't seem possible to use then() on setTimout but I changed the code to the following and it seems to work nicely.

`inputfun: function (args) {
      return new Promise((resolve, reject) => {
        setTimeout(function() {
          resolve(window.prompt(args))
          Sk.execStart = new Date();
        }, 250)
      })
    },`

Thanks for the help!
